ens/utils.py

```diff
@@ -108,16 +109,30 @@ def is_valid_name(name):
         return False
 
 
-def label_to_name(label, default_tld, recognized_tlds):
+def label_to_name(label, recognized_tlds):
     label = normalize_name(label)
     pieces = label.split('.')
     if pieces[-1] not in recognized_tlds:
-        pieces.append(default_tld)
+        raise InvalidTLD(
+            f"The label: {label} has an unsupported TLD of {pieces[-1]}. "
+            f"ENS.py by default supports the following TLDs: {recognized_tlds}. "
+            "If you'd like to use an unsupported TLD, please set the environment variable: "
+            "'ENS_RECOGNIZED_TLDS' to a string of desired TLDs separated by a colon (:)."
+        )
     return '.'.join(pieces)
 
 
 def dot_eth_name(label):
-    return label_to_name(label, 'eth', RECOGNIZED_TLDS)
+    recognized_tlds = get_recognized_tlds()
+    return label_to_name(label, recognized_tlds)
+
+
+def get_recognized_tlds():
+    if 'ENS_RECOGNIZED_TLDS' in os.environ:
+        override_tlds = os.environ['ENS_RECOGNIZED_TLDS'].split(':')
+        return set(DEFAULT_RECOGNIZED_TLDS + override_tlds)
+    else:
+        return DEFAULT_RECOGNIZED_TLDS
```

Comment:
@pipermerriam How about this? I wasn't sure how much checking of the env variable to do. (eg. even if it's an empty string, everything should still work fine - and the user should be able to tell where they went wrong via the error msg). I went for also including the default TLDs even if the user sets the env variable, not sure if this should be avoided or not?

Reply:
@pipermerriam Just saw this on the ens docs
> Can I register a TLD of my own in the ENS?
No, TLDs are restricted to only .eth (on mainnet), or .eth and .test (on Ropsten), plus any special purpose TLDs such as those required to permit reverse lookups. There are no immediate plans to invite proposals for additional TLDs. In large part this is to reduce the risk of a namespace collision with the IANA DNS namespace.

Do we still want to allow users to set their own tlds?

Reply:
For private networks I guess. Also for ENS team to test new TLDs. (I know Nick has used the library at least once, and requested changes) Also, that doc looks out of date. There is at least `.xyz` and I think another available now.